field/src/field.rs

```diff
@@ -146,10 +150,59 @@ pub trait PrimeCharacteristicRing:
     ///
     /// This function should be preferred over calling `a * a * a`, as a faster implementation may be available for some rings.
     #[must_use]
+    #[inline(always)]
     fn cube(&self) -> Self {
         self.square() * self.clone()
     }
 
+    /// Computes the arithmetic generalization of boolean `xor`.
+    ///
+    /// For boolean inputs, `x ^ y = x + y - 2 xy`.
+    #[must_use]
+    #[inline(always)]
+    fn xor(&self, y: &Self) -> Self {
+        self.clone() + y.clone() - self.clone() * y.clone().double()
+    }
+
+    /// Computes the arithmetic generalization of a triple `xor`.
+    ///
+    /// For boolean inputs `x ^ y ^ z = x + y + z - 2(xy + xz + yz) + 4xyz`.
+    #[must_use]
+    #[inline(always)]
+    fn xor3(&self, y: &Self, z: &Self) -> Self {
+        self.xor(y).xor(z)
+    }
+
+    /// Computes the arithmetic generalization of `andnot`.
+    ///
+    /// For boolean inputs `(!x) & y = (1 - x)y`.
+    #[must_use]
+    #[inline(always)]
+    fn andn(&self, y: &Self) -> Self {
+        (Self::ONE - self.clone()) * y.clone()
+    }
+
+    /// The zerofier of boolean values: `x * (1 - x)`.
+    ///
+    /// This is a polynomial of degree `2` evaluates to `0` if the input is `0` or `1`.
+    /// If our space is a field, then this will be non `0` on all other inputs.
+    #[must_use]
+    #[inline(always)]
+    fn bool_check(&self) -> Self {
+        // We use `x * (1 - x)` instead of `x * (x - 1)` as this lets us reuse the `andn` function.
+        self.andn(self)
+    }
+
+    /// The zerofier of ternary values: `x * (1 - x) * (2 - x)`.
```

Comment:
```suggestion
    /// The vanishing polynomial of the set {0, 1, 2}: `x * (1 - x) * (2 - x)`.
```
"Ternary" just means "consisting of three things", and in cryptography just as often refers to the set `{-1, 0, 1}` (e.g. in signed-digit representation of a ternary number system). For that reason this function should probably have a different name as the current one is ambiguous, though a better name doesn't immediately come to mind.

Reply:
Hmm, how about `trit` which is the analogue of a bit in ternary: (https://en.wikipedia.org/wiki/Ternary_numeral_system).

Reply:
I think those two names are equivalent, since neither resolves the ambiguity between the {0,1,2} and {-1,0,1} representations of ZZ/3 (both are referred to in the Wikipedia article you link to). Sorry, I'm being both pedantic and unhelpful. :sweat_smile: